groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -765,787 +1431,10708 @@ void Optional_DataImp<TYPE>::emplace(
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_02)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_03)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_04)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_05)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_06)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_07)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_08)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_09)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_10)
+                                   );
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_D >= 10
+
+#endif
+#else
+// The generated code below is a workaround for the absence of perfect
+// forwarding in some compilers.
+    template <class... ARGS>
+    void emplace(BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)...);
+
+#if defined(BSLS_COMPILERFEATURES_SUPPORT_GENERALIZED_INITIALIZERS)
+    template <class INIT_LIST_TYPE, class... ARGS>
+    void emplace(std::initializer_list<INIT_LIST_TYPE>,
+                                        BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)
+                                        ...);
+#endif
+// }}} END GENERATED CODE
+#endif
+    void reset() BSLS_KEYWORD_NOEXCEPT;
+        // Reset this object to the default constructed state (i.e., to a
+        // disengaged state).
+
+    void swap(optional& other);
+        // Efficiently exchange the value of this object with the value of the
+        // specified 'other' object.  In effect, performs
+        // 'using bsl::swap; swap(c, other.c);'.
+
+#ifdef BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+    TYPE&  value() &;
+    TYPE&& value() &&;
+        // Return a reference providing modifiable access to the underlying
+        // 'TYPE' object.  Throws a 'bsl::bad_optional_access' if the
+        // 'optional' object is disengaged.
+#else
+    TYPE& value();
+        // Return a reference providing modifiable access to the underlying
+        // 'TYPE' object.  Throws a 'bsl::bad_optional_access' if the
+        // 'optional' object is disengaged.
+
+#endif  // BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+#ifdef BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+    template <class ANY_TYPE>
+    TYPE value_or(ANY_TYPE&& rhs) &&;
+        // If 'this->has_value() == true', return a copy of the contained
+        // 'value_type' object.  Otherwise, return the value of specified 'rhs'
+        // converted to 'value_type'.
```

Comment:
In general, we try to phrase the purpose of the function in a single sentence.  `NullableValue` has a good reference in `valueOr`.

Reply:
copied and adapted the comment from nullableValue. I left in the note about it being inefficient. 
copied and adapted the comment form nullableValue in the allocator extended version of value_or. Forced the purpose into one sentence. Leaving the conversation unresolved until these new comments are looked at.